frontend/lib/resolution/scope-queries.cpp

```diff
@@ -571,13 +584,15 @@ bool LookupHelper::doLookupInImportsAndUses(
         auto scopeAst = parsing::idToAst(context, is.scope()->id());
         auto visibility = scopeAst->toDecl()->visibility();
         bool isMethodOrField = false;
-        bool onlyMethodsFields = false;
+        IdAndVis::SymbolTypeFlags filterFlags = 0;
+        if (!allowPrivateAccess) {
+          filterFlags |= IdAndVis::PUBLIC;
+        }
```

Comment:
I am understanding you to mean

```
if (!allowPrivateAccess) {
  filterFlags |= IdAndVis::PUBLIC;
} else {
  filterFlags |= IdAndVis::PRIVATE;
}
```

is that right?

That would not do what we want. When `allowPrivateAccess` is `true`, we should be able to find both public and private symbols. When it is `false`, we should be able to find only public symbols. There is no setting here where it should find only private symbols.

Keep in mind that in these filters, each setting is represented by two bits, `SOMETHING` and `NOT_SOMETHING`. I used `PUBLIC` and `PRIVATE` instead of `PUBLIC` and `NOT_PUBLIC` but that's just because we have a name for "not public" already. Anyway, if neither of these two bits are set, it is "don't care".